src/lib.rs

```diff
@@ -139,19 +138,10 @@ macro_rules! bitflags {
 
         impl ::std::fmt::Debug for $BitFlags {
             fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
-                let mut first = true;
-                $(
-                    if self.contains($Flag) {
-                        if !first {
-                            try!(f.write_str(" | "));
-                        }
-                        first = false;
-                        try!(f.write_str(stringify!($Flag)));
-                    }
-                )+
-                let _ = first; // silence! the last assignment from the above
-                               // expansion doesn't get read. fix that.
-                Ok(())
+                let out = format!("{} {{ bits: {:#010b} }}",
+                                  stringify!($BitFlags),
+                                  self.bits);
+                f.write_str(&out[..])
```

Comment:
General design note: there’s no need to create an intermediate string; you should use `write!` in here:

```rust
                write!(f, "{} {{ bits: {:#010b} }}",
                       stringify!($BitFlags),
                       self.bits)
```